File: src/util/entities/Ban.ts
import "reflect-metadata";
import { Column, Entity, JoinColumn, ManyToOne, Relation, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";
import { Guild } from "./Guild";
import { User } from "./User";

@Entity("bans")
export class Ban extends BaseClass {
	@Column({ nullable: true })
	@RelationId((ban: Ban) => ban.user)
	user_id: string;

	@JoinColumn({ name: "user_id" })
	@ManyToOne(() => User, {
		onDelete: "CASCADE"
	})
	user: Relation<User>;

	@Column({ nullable: true })
	@RelationId((ban: Ban) => ban.guild)
	guild_id: string;

	@JoinColumn({ name: "guild_id" })
	@ManyToOne(() => Guild, {
		onDelete: "CASCADE"
	})
	guild: Relation<Guild>;

	@Column({ nullable: true })
	@RelationId((ban: Ban) => ban.executor)
	executor_id: string;

	@JoinColumn({ name: "executor_id" })
	@ManyToOne(() => User)
	executor: Relation<User>;

	@Column()
	ip: string;

	@Column({ nullable: true })
	reason?: string;
}

File: extra/admin-api/Spacebar.Db/Models/Ban.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Db.Models;

[Table("bans")]
public partial class Ban
{
    [Key]
    [Column("id", TypeName = "character varying")]
    public string Id { get; set; } = null!;

    [Column("user_id", TypeName = "character varying")]
    public string? UserId { get; set; }

    [Column("guild_id", TypeName = "character varying")]
    public string? GuildId { get; set; }

    [Column("executor_id", TypeName = "character varying")]
    public string? ExecutorId { get; set; }

    [Column("ip", TypeName = "character varying")]
    public string Ip { get; set; } = null!;

    [Column("reason", TypeName = "character varying")]
    public string? Reason { get; set; }

    [ForeignKey("ExecutorId")]
    [InverseProperty("BanExecutors")]
    public virtual User? Executor { get; set; }

    [ForeignKey("GuildId")]
    [InverseProperty("Bans")]
    public virtual Guild? Guild { get; set; }

    [ForeignKey("UserId")]
    [InverseProperty("BanUsers")]
    public virtual User? User { get; set; }
}
